code-challenges.test.js

```diff
@@ -26,42 +26,110 @@ const numbersArray2 = [24, 27, 30, 33, 36]
-// Expected output: ["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"]
-const randomNouns2 = ["temperature", "database", "chopsticks", "mango"]
-// Expected output: ["Temperature", "Database", "Chopsticks", "Mango"]
-
+// describe("capArray", () => {
+//     it("takes in an array of words and returns an array with all the words capitalized.", () => {
+//         const randomNouns1 = ["streetlamp", "potato", "teeth", "conclusion", "nephew"]
+//         const randomNouns2 = ["temperature", "database", "chopsticks", "mango"]
+//       expect(capArray(randomNouns1)).toEqual(["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"])
+//       expect(capArray(randomNouns2)).toEqual(["Temperature", "Database", "Chopsticks", "Mango"])
+//     })
+//   }) //capArray › takes in an array of words and returns an array with all the words capitalized.
 // b) Create the function that makes the test pass.
-
+const capArray = (array) => {
+  let newArray = [];
+  for (let i = 0; i < array.length; i++) {
+    newArray.push(array[i].charAt(0).toUpperCase());
+  }
+  return newArray;
+}; // i got lost trying to find the issue. later i added .charAt method make sure it was the first lleter capitalized. Array [
+  // -   "Streetlamp",
+  // -   "Potato",
+  // -   "Teeth",
+  // -   "Conclusion",
+  // -   "Nephew",
+  // +   "S",
+  // +   "P",
+  // +   "T",
+  // +   "C",
+  // +   "N",
+  //   ]
 // Pseudo code:
+// create a function named capArray
+// it takes an array as an arument
+// use a for loop to iterate throught each index
+// capitalized each index with the .toUpperCase method
+// return the array
+// Input: an array of words
+// Output: an array with the strings capitalized
 
 // --------------------3) Create a function that takes in a string and logs the index of the first vowel.
-
 // a) Create a test with expect statements for each of the variables provided.
-
-const vowelTester1 = "learn"
-// Expected output: 1
-const vowelTester2 = "academy"
-// Expected output: 0
-const vowelTester3 = "challenges"
-// Expected output: 2
-
-// b) Create the function that makes the test pass.
+// describe("finFirstVowelIndex", () => {
+//     it("takes in a string and logs the index of the first vowel.", () => {
+//         const vowelTester1 = "learn"
+//         const vowelTester2 = "academy"
+//         const vowelTester3 = "challenges"
+//       expect(findFirstVowelIndex(vowelTester1)).toEqual(1)
+//       expect(findFirstVowelIndex(vowelTester2)).toEqual(0)
+//       expect(findFirstVowelIndex(vowelTester3)).toEqual(2)
+//     })
+//   }) // ● finFirstVowelIndex › takes in a string and logs the index of the first vowel.
+
+  // b) Create the function that makes the test pass.
+const findFirstVowelIndex = (str) => {
+  const vowels = ['a', 'e', 'i', 'o', 'u'];
+  const lowerCaseStr = str.toLowerCase();
+  
+  for (let i = 0; i < lowerCaseStr.length; i++) {
+    if (vowels.includes(lowerCaseStr[i])) {
+       return console.log(i);
```

Comment:
You are getting undefined because your function is not exactly returning anything. You have a return keyword, but there's no data that is getting outputted, there is only a console.log, which is only used to display something in the console. That being said, think about what you already have here, and what it is that you want your function to output. Very close!